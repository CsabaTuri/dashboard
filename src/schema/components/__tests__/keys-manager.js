import React from "react"
import { Map } from "immutable"
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"

import { reduceSchemaTree, KeysManager } from "../../"

describe("KeysManager Component", () => {
  let store

  beforeEach(() => {
    store = createImmutableTestStore({ schema: reduceSchemaTree })()
  })

  it("should render for root db", () => {
    const comp = shallow(<KeysManager store={store} />).dive()
    expect(shallowToJson(comp)).toMatchSnapshot()
  })

  it("should keys containment message when not a root database", () => {
    const db = Map.of(
      "isRoot", false,
      "name", "subdb",
      "parent", Map.of(
        "url", "/db/parent/databases"
      )
    )

    const comp = shallow(<KeysManager store={store} database={db} />).dive()
    expect(shallowToJson(comp)).toMatchSnapshot()
  })
})
