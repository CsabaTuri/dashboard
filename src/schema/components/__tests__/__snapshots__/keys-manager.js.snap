exports[`KeysManager Component should keys containment message when not a root database 1`] = `
<div>
  <Pivot>
    <PivotItem
      itemKey="keys"
      linkText="Keys">
      <p>
        Keys are located at the container database. If you want to grant access to 
        <strong>
          subdb
        </strong>
        , go to its 
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/db/parent/keys">
          parent
        </Link>
         container.
      </p>
      <Connect(KeysList) />
    </PivotItem>
    <PivotItem
      itemKey="create"
      linkText="Create a Key">
      <Connect(KeysForm) />
    </PivotItem>
  </Pivot>
</div>
`;

exports[`KeysManager Component should render for root db 1`] = `
<div>
  <Pivot>
    <PivotItem
      itemKey="keys"
      linkText="Keys">
      <Connect(KeysList) />
    </PivotItem>
    <PivotItem
      itemKey="create"
      linkText="Create a Key">
      <Connect(KeysForm) />
    </PivotItem>
  </Pivot>
</div>
`;
